src/pages/permissions/panels/EditGroupPermissionsPanel.tsx

```diff
@@ -0,0 +1,284 @@
+  const panelParams = usePanelParams();
+  const notify = useNotify();
+  const toastNotify = useToastNotification();
+  const queryClient = useQueryClient();
+  const [submitting, setSubmitting] = useState(false);
+  const [resourceType, setResourceType] = useState("");
+  const [resource, setResource] = useState("");
+  const [entitlement, setEntitlement] = useState("");
+  const [permissions, setPermissions] = useState<
+    (LxdPermission & { id: string })[]
+  >(() => initialiseGroupPermissions(group));
+  const [search, setSearch] = useState("");
+
+  const handleAddPermission = () => {
+    setPermissions((prev) => {
+      // we should prevent user from adding the same permission again
+      const currentId = resourceType + resource + entitlement;
+      const permissionExists = prev.some(
+        (permission) => permission.id === currentId,
+      );
+
+      if (permissionExists) {
+        return prev;
+      }
+
+      const isServerPermission = resourceType === "server";
+
+      return [
+        ...prev,
+        {
+          entity_type: resourceType,
+          url: isServerPermission ? "/1.0" : resource,
+          entitlement: entitlement,
+          id: currentId,
+        },
+      ];
+    });
+    setResourceType("");
+    setResource("");
+    setEntitlement("");
+  };
+
+  const handleDeletePermission = (permissionId: string) => {
+    setPermissions((prev) => {
+      const newPermissions = prev.filter(
+        (permission) => permission.id !== permissionId,
+      );
+      return newPermissions;
+    });
+  };
+
+  const handleSavePermissions = () => {
+    setSubmitting(true);
+    const groupPayload = {
+      name: group?.name,
+      description: group?.description,
+      permissions,
+    };
+
+    updateGroup(groupPayload)
+      .then(() => {
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.authGroups],
+        });
+        toastNotify.success(`Permissions for group ${group?.name} updated.`);
+        panelParams.clear();
+        notify.clear();
+      })
+      .catch((e) => {
+        notify.failure("Failed to update permissions", e);
+      })
+      .finally(() => {
+        setSubmitting(false);
+      });
+  };
+
+  const closePanel = () => {
+    panelParams.clear();
+    notify.clear();
+  };
+
+  const filteredPermissions = search
+    ? permissions.filter((permission) => {
+        return (
+          permission.entitlement.includes(search) ||
+          permission.entity_type.includes(search) ||
+          permission.url.toLowerCase().includes(search)
+        );
+      })
+    : permissions;
+
+  const headers = [
+    { content: "Resource type", sortKey: "resourceType" },
+    { content: "Resource", sortKey: "resource" },
+    { content: "Entitlement", sortKey: "entitlement" },
+    { "aria-label": "Actions", className: "u-align--right actions" },
+  ];
+
+  const rows = filteredPermissions.map((permission, idx) => {
```

Comment:
These rows seem to have no stable ordering. I think we should enforce an ordering in the ui: 

1. By resource type (same order as in the dropdown)
2. By displayed resource name
3. By entitlement

And probably we shouldn't let the user change this ordering in the panel.

Example of the random ordering experienced currently is below

![image](https://github.com/canonical/lxd-ui/assets/1155472/8b951bbc-7c08-4607-add7-8c8effe9f6fa)

Reply:
As discussed, we will have the above sorting order when the panel initially opens. The table is still sortable and adding a new permission will adhere to the currently active sorting criteria.

Reply:
Implemented the sorting order for this table as discussed